### PR TITLE
Support: Refactor course pages

### PR DESCRIPTION
**Pull request by paulrobertlloyd**
## Context

The current course page lists all applications and all vacancies, which makes it difficult to use (and I suspect can take a while to load).

This PR splits this page up into its component parts, and moves opening a course on Apply into it’s own page, accessible from a change link.

### Before

course-before

### After

course-details

course-applications

course-vacancies

course-open-on-apply

## Things to check

- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/components/support_interface/course_apply_status_component.html.erb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<%= render TagComponent.new(
```
Comment (TobyRet) [inline app/components/support_interface/course_apply_status_component.html.erb:1]: I'm not sure what the team policy is on component specs but I would imagine they would be good to have in instances where they are doing more than simply displaying html

File: app/views/support_interface/course/open_on_apply.html.erb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<% content_for :browser_title, 'Can candidates use Apply?' %>
```
Comment (TobyRet) [inline app/views/support_interface/course/open_on_apply.html.erb:1]: Breadcrumb required also on this page?

**Review by TobyRet — APPROVED** [review]
LGTM. Couple of minor things. I'll leave it up to you whether you decide to address in this PR or another one. 👍

File: app/components/support_interface/course_apply_status_component.html.erb (hunk 1 above)
Comment (paulrobertlloyd) [reply app/components/support_interface/course_apply_status_component.html.erb:1]: Would the alternative to be to have this in the view’s controller?

File: app/views/support_interface/course/open_on_apply.html.erb (hunk 2 above)
Comment (paulrobertlloyd) [reply app/views/support_interface/course/open_on_apply.html.erb:1]: Following a policy of using back links on pages like this (forms, flows, confirmation steps), and breadcrumbs on other pages where you’re browsing page within a deep hierarchy.

File: app/components/support_interface/course_apply_status_component.html.erb (hunk 1 above)
Comment (paulrobertlloyd) [reply app/components/support_interface/course_apply_status_component.html.erb:1]: @tvararu @duncanjbrown Thoughts? Is this component worthy, or should the code lie somewhere else?

Comment (duncanjbrown) [reply app/components/support_interface/course_apply_status_component.html.erb:1]: Matter of taste! I think I'd turn it into a single `if` with duplication in preference to two ternaries which quietly depend on each other?

**Review by tijmenb — COMMENTED** [review]
Could you check this with Josh? I strongly suspect that the Apply on/off setting is actually the only thing this page is used for, so moving this functionality would result in an extra click in 99% of journeys.

**Comment by tijmenb** [comment]
I've checked the data, and this page was used 328 times in he last 48 hours. In 88% of the time (290) the form was used, and in 12% of the time the page was only viewed, without an update.

**Comment by paulrobertlloyd** [comment]
@tijmenb Having spoken to Josh, have reverted details page to something closer to original layout, with the status setting at the bottom of the page. Not pushed below any vacancies or applications any more, should also be a little bit easier to update a course.

**Comment by paulrobertlloyd** [comment]
Update design for details tab:

Screenshot 2020-10-13 at 14 42 49

**Pull request merged**